### PR TITLE
test(#4117): add tests for null hash in FpIfReleased and related classes

### LINKED ISSUE #4117: `MjUnphi` fails with NPE
Recently, in one of my integration tests I got the following exception during the execution of `phi-to-xmir` goal:

```
[ERROR] Failed to execute goal org.eolang:eo-maven-plugin:0.56.0:phi-to-xmir (phi-to-xmir) on project phi-unphi-it: 'MjUnphi' execution failed: java.io.UncheckedIOException: java.io.IOException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Failed to process "D:\a\jeo-maven-plugin\jeo-maven-plugin\target\its\phi-unphi\target\generated-sources\eo-phi\org\eolang\hone\mess\D.phi" (sun.nio.fs.WindowsPath.WindowsPathWithAttributes): Cannot invoke "String.isEmpty()" because "hsh" is null -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eolang:eo-maven-plugin:0.56.0:phi-to-xmir (phi-to-xmir) on project phi-unphi-it: 'MjUnphi' execution failed
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.apache.maven.plugin.MojoFailureException: 'MjUnphi' execution failed
    at org.eolang.maven.MjSafe.exitError (MjSafe.java:624)
    at org.eolang.maven.MjSafe.execute (MjSafe.java:481)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: java.util.concurrent.ExecutionException: java.io.UncheckedIOException: java.io.IOException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Failed to process "D:\a\jeo-maven-plugin\jeo-maven-plugin\target\its\phi-unphi\target\generated-sources\eo-phi\org\eolang\hone\mess\D.phi" (sun.nio.fs.WindowsPath.WindowsPathWithAttributes)
    at java.util.concurrent.FutureTask.report (FutureTask.java:122)
    at java.util.concurrent.FutureTask.get (FutureTask.java:205)
    at org.eolang.maven.MjSafe.execWithTimeout (MjSafe.java:573)
    at org.eolang.maven.MjSafe.execute (MjSafe.java:463)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: java.io.UncheckedIOException: java.io.IOException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Failed to process "D:\a\jeo-maven-plugin\jeo-maven-plugin\target\its\phi-unphi\target\generated-sources\eo-phi\org\eolang\hone\mess\D.phi" (sun.nio.fs.WindowsPath.WindowsPathWithAttributes)
    at org.cactoos.scalar.Unchecked.value (Unchecked.java:39)
    at org.cactoos.iterator.Mapped$1.next (Mapped.java:49)
    at org.cactoos.iterator.IteratorEnvelope.next (IteratorEnvelope.java:40)
    at org.cactoos.scalar.Folded.value (Folded.java:63)
    at org.cactoos.number.NumberOfScalars.lambda$new$1 (NumberOfScalars.java:49)
    at org.cactoos.scalar.Checked.value (Checked.java:56)
    at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
    at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
    at org.cactoos.number.NumberOfScalars.intValue (NumberOfScalars.java:74)
    at org.cactoos.number.NumberEnvelope.intValue (NumberEnvelope.java:36)
    at org.eolang.maven.Threaded.total (Threaded.java:71)
    at org.eolang.maven.MjUnphi.exec (MjUnphi.java:109)
    at org.eolang.maven.MjSafe.lambda$execWithTimeout$6 (MjSafe.java:570)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
Caused by: java.io.IOException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Failed to process "D:\a\jeo-maven-plugin\jeo-maven-plugin\target\its\phi-unphi\target\generated-sources\eo-phi\org\eolang\hone\mess\D.phi" (sun.nio.fs.WindowsPath.WindowsPathWithAttributes)
    at org.cactoos.func.UncheckedFunc.lambda$apply$0 (UncheckedFunc.java:37)
    at org.cactoos.scalar.Checked.value (Checked.java:56)
    at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
    at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
    at org.cactoos.func.UncheckedFunc.apply (UncheckedFunc.java:38)
    at org.cactoos.scalar.Checked.wrappedException (Checked.java:80)
    at org.cactoos.scalar.Checked.value (Checked.java:65)
    at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
    at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
    at org.cactoos.iterator.Mapped$1.next (Mapped.java:49)
    at org.cactoos.iterator.IteratorEnvelope.next (IteratorEnvelope.java:40)
    at org.cactoos.scalar.Folded.value (Folded.java:63)
    at org.cactoos.number.NumberOfScalars.lambda$new$1 (NumberOfScalars.java:49)
    at org.cactoos.scalar.Checked.value (Checked.java:56)
    at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
    at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
    at org.cactoos.number.NumberOfScalars.intValue (NumberOfScalars.java:74)
    at org.cactoos.number.NumberEnvelope.intValue (NumberEnvelope.java:36)
    at org.eolang.maven.Threaded.total (Threaded.java:71)
    at org.eolang.maven.MjUnphi.exec (MjUnphi.java:109)
    at org.eolang.maven.MjSafe.lambda$execWithTimeout$6 (MjSafe.java:570)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Failed to process "D:\a\jeo-maven-plugin\jeo-maven-plugin\target\its\phi-unphi\target\generated-sources\eo-phi\org\eolang\hone\mess\D.phi" (sun.nio.fs.WindowsPath.WindowsPathWithAttributes)
    at java.util.concurrent.FutureTask.report (FutureTask.java:122)
    at java.util.concurrent.FutureTask.get (FutureTask.java:191)
    at org.cactoos.scalar.Mapped.lambda$new$0 (Mapped.java:39)
    at org.cactoos.scalar.ScalarEnvelope.value (ScalarEnvelope.java:34)
    at org.cactoos.scalar.Checked.value (Checked.java:56)
    at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
    at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
    at org.cactoos.iterator.Mapped$1.next (Mapped.java:49)
    at org.cactoos.iterator.IteratorEnvelope.next (IteratorEnvelope.java:40)
    at org.cactoos.scalar.Folded.value (Folded.java:63)
    at org.cactoos.number.NumberOfScalars.lambda$new$1 (NumberOfScalars.java:49)
    at org.cactoos.scalar.Checked.value (Checked.java:56)
    at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
    at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
    at org.cactoos.number.NumberOfScalars.intValue (NumberOfScalars.java:74)
    at org.cactoos.number.NumberEnvelope.intValue (NumberEnvelope.java:36)
    at org.eolang.maven.Threaded.total (Threaded.java:71)
    at org.eolang.maven.MjUnphi.exec (MjUnphi.java:109)
    at org.eolang.maven.MjSafe.lambda$execWithTimeout$6 (MjSafe.java:570)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
Caused by: java.lang.IllegalStateException: Failed to process "D:\a\jeo-maven-plugin\jeo-maven-plugin\target\its\phi-unphi\target\generated-sources\eo-phi\org\eolang\hone\mess\D.phi" (sun.nio.fs.WindowsPath.WindowsPathWithAttributes)
    at org.eolang.maven.Threaded.lambda$total$0 (Threaded.java:60)
    at org.cactoos.scalar.CallableEnvelope.call (CallableEnvelope.java:34)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
Caused by: java.lang.NullPointerException: Cannot invoke "String.isEmpty()" because "hsh" is null
    at org.eolang.maven.FpIfReleased.lambda$new$1 (FpIfReleased.java:56)
    at org.cactoos.func.UncheckedBiFunc.lambda$apply$0 (UncheckedBiFunc.java:38)
    at org.cactoos.scalar.Checked.value (Checked.java:56)
    at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
    at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
    at org.cactoos.func.UncheckedBiFunc.apply (UncheckedBiFunc.java:39)
    at org.eolang.maven.FpFork.lambda$new$1 (FpFork.java:39)
    at org.eolang.maven.FpEnvelope.apply (FpEnvelope.java:31)
    at org.eolang.maven.FpEnvelope.apply (FpEnvelope.java:31)
    at org.eolang.maven.FpExistedSource.lambda$new$0 (FpExistedSource.java:25)
    at org.eolang.maven.FpEnvelope.apply (FpEnvelope.java:31)
    at org.eolang.maven.FpEnvelope.apply (FpEnvelope.java:31)
    at org.eolang.maven.MjUnphi.lambda$exec$1 (MjUnphi.java:101)
    at org.eolang.maven.Threaded.lambda$total$0 (Threaded.java:56)
    at org.cactoos.scalar.CallableEnvelope.call (CallableEnvelope.java:34)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
```

It looks like a flaky behaviour because the same test had passed for other OS (like Linux, MacOS) and then it failed for Windows platform only.

I used `org.eolang:eo-maven-plugin:0.56.0:phi-to-xmir (phi-to-xmir)`
Phi file that I was trying to UNPHI:
[D.phi.txt](https://github.com/user-attachments/files/19868774/D.phi.txt)

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
Originally, I was looking for the cause of the #4117 issue. However, it turned out, that we have already solved it [here](https://github.com/objectionary/eo/pull/4104). The solution is provided in the `0.56.2` version.

This problem used to happen because `ChCached` returned `null` values due concurrency issues. Since we solved the original problem with `null` values, we also solved this one. 

In this PR I only add tests for `null` hash parameters  (I used them to find a real cause of the issue.)

Closes #4117

### CODE REVIEW

File: eo-maven-plugin/src/test/java/org/eolang/maven/FpIfReleasedTest.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * Test case for {@link FpIfReleased}.
+ * @since 0.57
+ */
+final class FpIfReleasedTest {
+
+    @Test
+    void throwsNpeIfHashIsNull(@TempDir final Path tmp) {
+        Assertions.assertThrows(
+            NullPointerException.class,
```
[the remark this answers is not in the document]
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/FpIfReleasedTest.java:21]: @yegor256 This test just explains that we throw NPE in this particular case. I didn't have an intention to change the production code. But I can add a NPE check and throw `IllegalArgumentException` or `IllegalStateException` for this case. However, it might be considered as [not a good idea](https://www.yegor256.com/2014/05/13/why-null-is-bad.html#ad-hoc-error-handling). What do you think?

**Review by yegor256 — APPROVED** [review]

**Comment by 0crat** [comment]
@volodya-lombrozo Thank you for your contribution! We've reviewed your code and awarded you +16 points, which aligns with our policy for code contributions. Your [hits-of-code](https://www.yegor256.com/2014/11/14/hits-of-code.html) and review process met our standards. We appreciate your efforts in maintaining code quality and encourage you to [keep](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) up the good work. Your updated running balance is now [+51](https://www.eolang.org/zerocracy/objectionary-vitals.html). Keep contributing and striving for excellence in your code!

**Comment by 0crat** [comment]
@yegor256 Thanks for your review! Here's how your +4 points were calculated: +12 base, -10 for having only 3 comments (policy requires 6+), and +2 to meet the minimum 4-point reward. While the review was brief, every contribution counts! Your current balance stands at [-774](https://www.eolang.org/zerocracy/objectionary-vitals.html). Keep up the good work and aim for more detailed reviews next time to maximize your points! 💻✨
